### PR TITLE
Complete environment description

### PR DESCRIPTION
**Pull request by daniel-shimon**
Shows the env var's description when completing it 🔥 
Also show its type in the display:

image

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

File: xonsh/completers/environment.py [hunk 1]
```diff
@@ -27,4 +28,8 @@ def complete_environment_vars(context: CompletionContext):
     filter_func = get_filter_function()
     env_names = XSH.env
 
-    return {"$" + k for k in env_names if filter_func(k, key)}, lprefix
+    return {
+        RichCompletion("$" + k, description=env_names.get_docs(k).doc)
+        for k in env_names
+        if filter_func(k, key)
+    }, lprefix
```
Comment (jnoortheen) [inline xonsh/completers/environment.py:35]: perhaps a generator. we had some discussion on ordering

**Comment by gforsyth** [comment]
This is awesome @daniel-shimon!  
This may not be possible but I noticed that for our longer descriptions, the helper text at the bottom of the completion box doesn't wrap (and clips off the edge of the terminal).

File: xonsh/completers/environment.py (hunk 1 above)
Comment (daniel-shimon) [reply xonsh/completers/environment.py]: Done

**Comment by daniel-shimon** [comment]
@gforsyth Sadly this is an issue with upstream ptk. As filed by @jnoortheen in https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1285

**Comment by gforsyth** [comment]
> @gforsyth Sadly this is an issue with upstream ptk. As filed by @jnoortheen in https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1285

Still awesome! And we can also probably spend some time rewording some descriptions.

**Review by jnoortheen — APPROVED** [review]

**Pull request merged**